src/wps/core/repository.py

```diff
@@ -230,42 +353,44 @@ async def get(
             raise access_error
 
         if check_valid:
-            if not work_package.created <= now_as_utc() <= work_package.expires:
-                access_error = self.WorkPackageAccessError("Work package has expired")
-                log.error(access_error, extra=extra)
-                raise access_error
-
-            if work_package.type != WorkType.DOWNLOAD:
-                access_error = self.WorkPackageAccessError("Unsupported work type")
-                log.error(access_error, extra=extra)
-                raise access_error
-
-            if not await self._access.check_download_access(
-                work_package.user_id, work_package.dataset_id
-            ):
-                access_error = self.WorkPackageAccessError("Access has been revoked")
+            self._validate_work_package(work_package)
+
+            # Check access based on work package type
+            if work_package.type == WorkPackageType.DOWNLOAD:
+                has_access = await self._access.check_download_access(
+                    work_package.user_id,
+                    work_package.dataset_id,  # type: ignore
+                )
+            elif work_package.type == WorkPackageType.UPLOAD:
+                has_access = await self._access.check_upload_access(
+                    work_package.user_id,
+                    work_package.box_id,  # type: ignore
+                )
```

Comment:
```suggestion
                )
            else:  # pragma: no cover
                has_access = False
```

just in case and to make linters happy